source/postcard-c/postcard.h

```diff
@@ -0,0 +1,1184 @@
+#ifndef POSTCARD_H
+#define POSTCARD_H
+
+#include <stdbool.h>
+#include <stddef.h>
+#include <stdint.h>
+#include <string.h>
+
+/// A growable view into a buffer
+typedef struct {
+    // Pointer to the underlying data buffer
+    uint8_t* data;
+    /// The total written length of data in the buffer
+    size_t len; // Current length of data
+    /// The maximum capacity of the buffer
+    size_t capacity;
+} postcard_slice_t;
+
+/// postcard_c error codes
+typedef enum {
+    POSTCARD_SUCCESS = 0,
+    POSTCARD_ERROR_BUFFER_TOO_SMALL,
+    POSTCARD_ERROR_INVALID_INPUT,
+    POSTCARD_ERROR_INCOMPLETE_DATA,
+    POSTCARD_ERROR_OVERFLOW
+} postcard_error_t;
```

Comment:
I would prefer each error code to get an explicitly defined value. Whether the error values are positive or negative values is a different question.